lib/gltfPipeline.js

```diff
@@ -26,13 +29,24 @@ module.exports = {
     processFileToDisk : processFileToDisk
 };
 
-function processJSON(gltf, options, callback) {
+function writeSources(gltf, outputPath, options, callback) {
+    var embed = options.embed;
+    var embedImage = options.embedImage;
+    var basePath = path.dirname(outputPath);
+    async.each(['buffers', 'images', 'shaders'], function(name) {
+        writeSource(gltf, basePath, name, embed, embedImage, callback);
+    });
```

Comment:
You need a callback for when the async finishes, and instead you should send in `asyncCallback` to `writeSource`. Check out how `writeGltf` handles this.